dvc/repo/metrics/show.py

```diff
@@ -188,10 +188,16 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
         if not o.metric:
             continue
 
+        # NOTE this case assumes that metric has been modified with
+        # `dvc metric modify -t {} -x {}`, so stage file should contain
+        # info what type metric is and what is its xpath
         if not typ and isinstance(o.metric, dict):
```

Comment:
@pared @efiop I'm not sure I understand the comment tbh. Can you guys clarify it a little bit? Am I correct that `typ` variable is actually coming from the command line. In this case I still don't understand then how all these CLI options and DVC-file specified values played before and play now together.

So, if `typ` is defined then it will always take `type` and `xpath` from the command line, no matter what is defined in the DVC-file.

Reply:
Yes, typ is coming from the command line. And also, it is true that it can overwrite what is written inside dvc-file. Maybe we should not do that.